### PR TITLE
Add Gaudi support

### PR DESCRIPTION
**Pull request by yongfengdu**
## Description
Added Gaudi support for tgi server.
Added proxy support.
Updated container image name with latest changes.
Added manifest for Gaudi.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Tests

Verified each services works fine.

### CODE REVIEW

File: helm-charts/codegen/gaudi-values.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  image:
+    repository: opea/llm-tgi:latest
+  service:
+    port: 9000
+# To override values in subchart tgi
+  tgi:
+    # LLM_MODEL_ID: ise-uiuc/Magicoder-S-DS-6.7B
+    # LLM_MODEL_ID: /data/Magicoder-S-DS-6.7B
+    LLM_MODEL_ID: /data/OpenCodeInterpreter-DS-6.7B
+    volume: /mnt/data0/weights
```
Comment (daisy-ycguo) [inline helm-charts/codegen/gaudi-values.yaml:33]: user might not have this folder which will cause error

File: helm-charts/codegen/gaudi-values.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  HUGGINGFACEHUB_API_TOKEN: "insert-your-huggingface-token-here"
+  image:
+    repository: opea/llm-tgi:latest
+  service:
+    port: 9000
+# To override values in subchart tgi
+  tgi:
+    # LLM_MODEL_ID: ise-uiuc/Magicoder-S-DS-6.7B
+    # LLM_MODEL_ID: /data/Magicoder-S-DS-6.7B
+    LLM_MODEL_ID: /data/OpenCodeInterpreter-DS-6.7B
```
Comment (daisy-ycguo) [inline helm-charts/codegen/gaudi-values.yaml:32]: too many commented lines, better to remove one.

**Review by daisy-ycguo — CHANGES_REQUESTED** [review]

**Review by daisy-ycguo — APPROVED** [review]
LGTM

File: helm-charts/codegen/gaudi-values.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    volume: /mnt/data0/weights
+    image:
+      repository: ghcr.io/huggingface/tgi-gaudi
+      tag: "1.2.1"
+    resources:
+      limits:
+        habana.ai/gaudi: 1
+global:
```
Comment (jfding) [inline helm-charts/codegen/gaudi-values.yaml:39]: please remove below lines

Comment (jfding) [reply helm-charts/codegen/gaudi-values.yaml:39]: or use <placeholder> msg here

File: helm-charts/codegen/gaudi-values.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+# To override values in subchart llm-uservice
+llm-uservice:
+  HUGGINGFACEHUB_API_TOKEN: "insert-your-huggingface-token-here"
+  image:
+    repository: opea/llm-tgi:latest
+  service:
+    port: 9000
+# To override values in subchart tgi
+  tgi:
+    # LLM_MODEL_ID: ise-uiuc/Magicoder-S-DS-6.7B
```
Comment (jfding) [inline helm-charts/codegen/gaudi-values.yaml:30]: clean up comments lines

**Review by jfding — CHANGES_REQUESTED** [review]

File: helm-charts/codegen/gaudi-values.yaml (hunk 4 above)
Comment (yongfengdu) [reply helm-charts/codegen/gaudi-values.yaml:30]: Would prefer to keep this comment as examples for user.
Comment line is a different format to specify model.

**Review by jfding — APPROVED** [review]

**Pull request merged**